internal/wrphandlers/qos/options.go

```diff
@@ -45,3 +45,14 @@ func MaxMessageBytes(s int) Option {
 			return nil
 		})
 }
+
+// PrioritizeOldest determines whether to prioritize the oldest message during a QualityOfService tie breaker,
+// with the default being to prioritize the newest messages.
+func PrioritizeOldest(b bool) Option {
+	return optionFunc(
+		func(h *Handler) error {
+			h.prioritizeOldest = b
```

Comment:
Instead of setting a boolean here, use a function that does the comparison.  That will let you remove a few conditional `if` statements elsewhere.